### ISSUE TITLE
The validation mask in validation and running an inference it's not similar

### ISSUE THREAD

**Issue by Cetvi** [MIC-DKFZ/nnUNet#2787]
I'm running an inference from a pretrained model that i did. The thing is that when the model is in training mode it performs a validation which is stored in the folder "nnUNetTrainer__nnUNetPlans__3d_fullres\fold_3\validation" i was confuse because the model was a good one, but when i run an inference the result is very bad. So i decided to run an inference with the images with which the model has done the validation phase, however, the result is not even similar. I don't know what i'm doing wrong because i'm running both with the same images, with the same trainer, with the same plans and the normalization_schemes is "NoNormalization.

**Comment by MartinCartajena** [comment]
Same issue here with nnUNetv2: validation looks great, but nnUNetv2_predict on the same images/checkpoint is much more worse. I haven’t found the cause — did you manage to solve it or find a workaround?

**Comment by seziegler** [comment]
Hi, sorry for the late reply. If this is still relevant for you, did your input images were the already preprocessed ones? `nnUNetv2_predict` expects raw images, applies the preprocessing on the fly and does the prediction. So it could be that you applied some preprocessing twice? Also, tta is applied to the final val predictions after training as well, so you shouldn't disable this to obtain same results.

**Comment by jgoco** [comment]
I'm still having the same problem, has anyone found a solution?

**Comment by seziegler** [comment]
Hi @jgoco , are you using the predict command on your raw or preprocessed folder? As I said earlier, `nnUNetv2_predict` is expecting raw images and will apply preprocessing during inference. When passing already preprocessed data (like data that you use for validation during training) you will apply preprocessing twice and will get worse results.

**Comment by jgoco** [comment]
Hi @seziegler, thanks for getting back to me. I predicted on the images in the raw folder using the following:
`nnUNetv2_predict -i "INPUT_FOLDER" -o "OUTPUT_FOLDER" -d 004 -c 2d -f 0`

**Comment by seziegler** [comment]
Hi @jgoco , so in the input folder you have the raw, non-preprocessed validation images of fold 0? Did you maybe use any custom trainer or custom plans? If so, you would need to specify those also for the predict command using the `-tr` and `-p` flags. Please also see [here](https://github.com/MIC-DKFZ/nnUNet/blob/master/documentation/how_to_use_nnunet.md#run-inference) if you haven't yet and check if your input data complies with the format requirements. Can you please share the command you used for training? Maybe I can spot something there that could explain it.

**Comment by jgoco** [comment]
Hi @seziegler, yup, the input folder contains raw, non-preprocessed validation images. I didn't use any custom trainer or plans to my knowledge. Here are the additional command I used:
`nnUNetv2_plan_and_preprocess -d 004 --verify_dataset_integrity`
`nnUNetv2_train 004 2d 0`

**Comment by seziegler** [comment]
Hi @jgoco , thanks for your answer! From your commands it looks like you've done everything correct. I'm not really sure what went wrong. Would it be feasible to rerun your preprocessing, training and inference in a new clean environment to check if the problem persists? Make sure you have the latest nnunet version installed. Maybe you could do this just for verification e.g. with a fraction of the data or simply using less epochs during training. If the problem is reproducible you can share a fraction of your dataset if you want for me to run it myself and see if I can spot something.
